content/articles/web-automation-with-selenium/index.md

```diff
@@ -0,0 +1,107 @@
+In this guide, we will be looking at how to use selenium to write scripts that will automate basic web tasks using python to give you a background of how Selenium works.
+### Prerequisite
+To understand this guide the reader must be familiar with HTML syntax such as; tags, elements, Ids, and classes. It is also assumed the reader is familiar with the basics of python programming language, concepts such as its syntax and creating a virtual environment are required.
+### Goal
+In this guide, we will be writing a python script that will perform a search on a browser. At the end of this guide the reader will be able to write a python script that can;
+- Find elements in the browser, 
+- send a text to form fields in the browser,
+- Click buttons in the browser
+
+### Setting up the environment
+First, we will need to create a virtual environment for its best practice to create a virtual environment for every python project we want to build so we don’t have issues with our project in the future. Click [here](https://uoa-eresearch.github.io/eresearch-cookbook/recipe/2014/11/26/python-virtual-env/) to learn how to create a virtual environment if you don’t know how to create one.
+To work with selenium we will have to install selenium, we will do that using the command below
+```
+pip install selenium
+```
+We also have to install a webdriver which is a tool that is needed for web automation. The web driver helps us interact with the browser. If you are using windows We will be using a windows package manager known as chocolatey to install the web driver, click [here]( https://chocolatey.org/install) to install chocolatey. To install we will use the command below
+```
+choco install chromdriver
+```
+If you are on a mac you the below command will do
+```
+brew cask install chromdriver
+```
+The chromedriver version should be compactable with your browser version but if in any way you encounter a compactable error, go to this [site](https://chromedriver.chromium.org/downloads) to download for your browser version and replace the one you have.
+
+Create a file `app.py` or whatever you will want to call your app and update it with the code below.
+
+```python
+from selenium import webdriver
+
+driver = webdriver.Chrome()
+
+driver.get('https://www.google.com/')
+```
+The above code snippet is used to open a browser and request a web `url`. The first line of code is importing webdriver from Selenium, in the second line we used the webdriver to open chrome which we stored in a variable `driver`.
+
+NOTE: There are different webdrivers for different browsers so if you prefer another browser just look for the webdriver for that browser on the internet, for instance, we would use `firefoxDriver` for the firefox browser. We are opening Chrome because we are using the `ChromeDriver`.
```

Comment:
```suggestion
> NOTE: There are different webdrivers for different browsers so if you prefer another browser just look for the webdriver for that browser on the internet, for instance, we would use `firefoxDriver` for the firefox browser. We are opening Chrome because we are using the `ChromeDriver`.
```